src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -1450,10 +1450,8 @@ private void redirectOrRejectRequest(HttpServletRequest req, HttpServletResponse
             throws IOException, ServletException {
         if (req.getSession(false) != null || Strings.isNullOrEmpty(req.getHeader("Authorization"))) {
             req.getSession().invalidate();
-            res.sendRedirect(Jenkins.get().getSecurityRealm().getLoginUrl());
-        } else {
-            res.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Token expired");
         }
+        res.sendRedirect(Jenkins.get().getSecurityRealm().getLoginUrl());
```

Comment:
The name of the function is now misleading.

I have to tack down where it is used, I would want the user to be stuck in a loop.

Reply:
OK. It is used only in one place (just above in fact).
The redirect does make sense.

If you change the name of the function, it is good to me.